trl/trainer/utils.py

```diff
@@ -1807,6 +1813,123 @@ def entropy_from_logits(logits, chunk_size: int = 1) -> torch.Tensor:
     return per_token_entropies
 
 
+def validate_and_preprocess_images(images: list, processing_class=None) -> list:
```

Comment:
At this stage, having a function that tries to handle all formats really doesn't seem like a good idea. There will always be a case you haven't considered, and that requires a great deal of maintenance to have this in the codebase. I think it's up to the user to comply with the supported format, even if it means pre-processing the dataset.